### ISSUE TITLE
Add ID attribute to root element

### ISSUE THREAD

**Issue by davecardwell** [tailwindlabs/heroicons#103]
Would there be any support for adding an ID with the icon name to the root `<svg>` element?

This would allow for referencing the icons in an SVG `<use>` element. For example, adding `id="hi-archive"` would allow importing via:

```html
<svg class="h-6 w-6 text-red-500">
  <use xlink:href="/img/heroicons/outline/archive.svg#hi-archive"/>
</svg>
```

This would be ideal for things like menu icons that are used on every page, eliminating the need to embed the full SVG each time while still allowing them to have the color styled.

It would increase the file size of the SVGs, but tools like svgo’s `cleanupIDs` can be used to mitigate this if you’re not using them.

**Comment by Th3S4mur41** [comment]
Yes please, this would be great to create SVG Sprites and take better advantage of caching

**Comment by davecardwell** [comment]
If it helps anybody else in the meantime I am using the following [SVGO](https://github.com/svg/svgo) “plugin” to add an ID to the `<svg>` element:

```js
plugins: [
  {
    // add id="root" to the <svg/> element so we can import with
    // <use xlink:href="foo.svg#root" />
    addAttributesToSVGElement: {
      attributes: [{ id: "root" }],
    },
  },
],
```

I would still prefer it to be included in the set itself if possible as I wouldn’t otherwise need to use SVGO in this project, and don’t always have an image pre-processing step in others.

**Comment by Th3S4mur41** [comment]
@davecardwell wouldn't it make more sense to use both folder and filename as id?

e.g. `heroicons/outline/archive.svg` should get the ID `outline-archive`
That way you can also combine them in a sprite, which would not work with the same hardcoded ID for each image.

**Comment by davecardwell** [comment]
@Th3S4mur41 that would be a more generally usable solution, yes—good idea.

For my current project I’m using `<use xlink:href="…">` for single SVGs so having the same ID works.

**Comment by Th3S4mur41** [comment]
Found a solution through webpack configuration that doesn't require editing the assets...
The following config in my Vue project add automatically the file name as ID when creating the sprite
```
configureWebpack: {
		module: {
			rules: [
				{
					test: /\.svg$/,
					use: ['svg-sprite-loader', 'svgo-loader']
				}
			]
		}
	},
chainWebpack: (config) => {
		config.module
			.rule('svg')
			.test(() => {
				return false;
			})
			.use('file-loader');
	}
```
That makes using any icon from the sprite possible with just `<svg><use href="#icon_name" /></svg>`
